src/vsphere_cpi/lib/cloud/vsphere/resources/disk.rb

```diff
@@ -13,7 +15,7 @@ def initialize(size_in_mb, datastore, folder, cid)
       end
 
       def path
-        "[#{@datastore.name}] #{@folder}/#{@cid}.vmdk"
+        "[#{URI.escape(@datastore.name)}] #{URI.escape(@folder)}/#{URI.escape(@cid)}.vmdk"
```

Comment:
This change really requires a link to some documentation or otherwise. I'm not sure if the string inside of the datastore name `[]` part of the path can be URL escaped.

Reply:
@ktchen14  Here is an escaped example `"[isc-cl1-ds-0] %CE%B1%CE%B2_%C3%85%C3%85%C3%85%C3%85_4e4ae49d16fa/ephemeral_disk.vmdk"` For here, path is `αβ_ÅÅÅÅ_4e4ae49d16fa`